src/lib.rs

```diff
@@ -467,14 +465,21 @@ fn expand_template(
     Ok(())
 }
 
+/// Try to add all provided template_values to the liquid_object.
+///
+/// ## Note:
+/// Values for which a placeholder exists, should already be filled by `fill_project_variables`
 pub(crate) fn add_missing_provided_values(
-    mut liquid_object: liquid::Object,
+    liquid_object: Rc<RefCell<liquid::Object>>,
```

Comment:
why would we need and want to introduce a `Rc<RefCell>` here?
I think so far we made a good job avoiding any smart pointers, let's keep that spirit as hard as we can.

Reply:
That's simple my friend; I was lazy, and didn't want to clone due to the `'static` requirements from the `rhai` api.

fix comming up!